cdk/setup-tool

```diff
@@ -0,0 +1,252 @@
+#!/bin/bash
+
+MY_VERSION="1.0"
+MY_NAME="acmne-setup-tool"
+MY_DESCRIPTION="ACM for Nitro Enclaves Setup Tool"
+
+# Enhanced utility functions
+say() {
+    [ -t 1 ] && [ -n "$TERM" ] \
+        && echo "$(tput setaf 2)[$MY_NAME]$(tput sgr0) $*" \
+        || echo "[$MY_NAME] $*"
+}
+
+say_err() {
+    [ -t 2 ] && [ -n "$TERM" ] \
+        && echo "$(tput setaf 1)[$MY_NAME] $*$(tput sgr0)" 1>&2 \
+        || echo "[$MY_NAME] $*" 1>&2
+}
+
+die() {
+    local code=1
+    [[ "$1" = "-c" ]] && {
+        code="$2"
+        shift 2
+    }
+    [[ $code -ne 0 ]] && say_err "$@"
+    exit $code
+}
+
+ok_or_die() {
+    local code=$?
+    [[ $code -eq 0 ]] || die -c $code "$@"
+}
+
+ensure_npx_installed() {
+    which npx > /dev/null 2>&1
+    ok_or_die "NPX cannot be found. Aborting." \
+        "Please make sure you have NPX installed."
+}
+
+ensure_cdk_cli_installed() {
+    which cdk > /dev/null 2>&1
+    ok_or_die "The CDK CLI cannot be found. Aborting." \
+        "Please make sure you have AWS CDK CLI installed."
+}
+
+validate_setup_name() {
+    local setup_name="$1"
+    if [[ ! "$setup_name" =~ ^[a-zA-Z0-9-]+$ ]]; then
+        die "Invalid setup name. Use only alphanumeric characters and hyphens."
+    fi
+    if [[ ${#setup_name} -gt 128 ]]; then
+        die "Setup name too long. Maximum length is 128 characters."
+    fi
+}
+
+# Deploy command implementation
+run_deploy_command() {
+    local setup_name=""
+    local args="$@"
+    
+    # Extract setup name from arguments if present
+    while [[ $# -gt 0 ]]; do
+        case "$1" in
+            --setup-name)
+                setup_name="$2"
+                validate_setup_name "$setup_name"
+                shift 2
+                ;;
+            *)
+                shift
+                ;;
+        esac
+    done
+    
+    [[ -n "$setup_name" ]] && say "Deploying setup: ${setup_name}"
+    
+    CLI_COMMAND="npx ts-node --prefer-ts-exts src/bin/cli.ts ${args}"
+    
+    cdk deploy --all --require-approval never --app \""${CLI_COMMAND}"\"
```

Comment:
`--require-approval never` hidden from the unassuming user. The command may change security posture of resources in their AWS account, so please do not blanket approve all those changes without the user knowing or having a choice.